### ISSUE TITLE
Very slow startup times

### ISSUE THREAD

**Issue by TheWattyk** [fastaddons/GroupSpeedDial#330]
The same as issue #123 
Only difference I'm using Mozilla Firefox 115.11.0esr (64-bit) instead of Chrome.

**Comment by Juraj-Masiar** [comment]
The startup time at that time was caused by Chromium issue when the addon database contained long strings (thumbnails were stored as text for compatibility reasons).
I've then released update that replaced the strings with "Blob" files which 100% fixed the issues and since then, I haven't received any more reports.

Note that recently there have been some reports of slow Firefox, although not related to my addon:
https://bugzilla.mozilla.org/show_bug.cgi?id=1899015

Try to backup your dials and thumbnails to a file, then uninstall my addon, then restart Firefox and see if helped. 
Then install it back and import your exported file. More info here:
https://github.com/fastaddons/GroupSpeedDial/wiki/Fixing-broken-dials-page#what-if-restarting-pc-didnt-helped

**Comment by arbaev** [comment]
Hi!

For some time now, GSD has also started to take a long time to start. I attributed this to the fact that I use too many dials. Today I got around to deleting GSD and doing a clean default installation. The result is that the lag remains, alas! And this is only with the default dialer group - an Examples.
When disabling GSD, Firefox starts almost instantly. I even disabled all other extensions to avoid conflicts with them. No effect.

I recorded a short video demonstrating the delay: https://monosnap.com/file/qouhM55QA0PbsRY2qSzBTIe2P7GBaC

What could be the problem? 🤔

My setup:
Firefox v127.0.2
Windows 10 Pro 22H2 19045.2728
GSD 25.2

**Comment by Juraj-Masiar** [comment]
Hello @arbaev ,
That's insane, am I seeing it righ? It's that slow even without imported data?
But the addon is empty, it cannot be slow :).
This is highly unlikely caused by my addon, we can test that by trying a different addon that overrides new tab page, for example:
https://addons.mozilla.org/en-US/firefox/addon/new-tab-override/
https://addons.mozilla.org/en-US/firefox/addon/tabliss/

Also, could you open your Windows Task Manager, switch to Performance tab and look - while Firefox starts - which PC part is being used, for example is disk 100%? Or maybe one of the CPU?

Regarding fixing, since this is most likely issue in your Firefox profile, the best solution is to reset it, more info:
https://support.mozilla.org/kb/refresh-firefox-reset-add-ons-and-settings

**Comment by arbaev** [comment]
Hi @Juraj-Masiar!
Most likely you are right, the problem is in my Firefox profile, it is very old. I am not ready to refresh the profile now, but installing Tabliss confirmed the assumption - with this addon Firefox also starts slowly.
I recorded a video: https://monosnap.com/file/otuc7kxayOVhRUDsSo2VAXLEAOhSHy
Yeah, the SSD is loaded at 99% for 10 seconds. Then I cleared the entire browser cache (2.6GB), now start-up time takes about 3 seconds. Well...

But now it is clear that the problem is not in GSD. Thank you for your answer! 🤝

**Comment by Juraj-Masiar** [comment]
Thanks for the update!
It's most likely caused by the bug I've linked above, or one of those other linked in that bug:
https://bugzilla.mozilla.org/show_bug.cgi?id=1778472
https://bugzilla.mozilla.org/show_bug.cgi?id=1899015
It seems that some storage initialization takes way too long and then addons that uses synchronous storage (like localStorage) will be blocked until it's finished.
The good news is, they increased priority for those bugs so it's more likely it will get fixed.
Until then, clearing cache should help a bit.

**Comment by andfinally** [comment]
I wonder if this might happen when there's a pending Firefox update?

I was noticing quite a long delay before anything loaded on the new tab page this morning on macOS FF 153.0.3 (aarch64). The page was just black. A browser restart didn't help. But when I went to check the FF version, it started downloading an update. After installing the update to 153.0.4 (aarch64) and restarting, the tab page loads immediately again.

**Comment by Juraj-Masiar** [comment]
Yeah, I've seen it too, maybe a month ago. And it also disappeared after restarting Firefox.
But I don't think it's related to updates. For me it happened after Firefox was opened for a long time, like several days (I'm using sleep/hibernate so my apps are running for days/weeks). And it was somewhat inconsistent, but the more new tabs I've opened (in a row), the slower they loaded.

I've seen it only like 3 times though, and I've tried to find out what is causing the delay, but I couldn't find anything (I've checked all logs and consoles). Now that I think about, I should have use the Firefox Profiler, oh, that would likely help find the real cause.

Anyway, there was a [database issue in Firefox 149](https://bugzilla.mozilla.org/show_bug.cgi?id=2044020) that got [fixed in current Firefox 153](https://bugzilla.mozilla.org/show_bug.cgi?id=1849842).
Oh, but I see now you already have 153 version, so this will be something else.
I'll investigate this more when I see it next time.

**Comment by andfinally** [comment]
Thanks @Juraj-Masiar, that could also have been the case for me in this instance – I tend to leave FF open a long time.
